### PR TITLE
Add peak values for all input devices in selection dialog

### PR DESCRIPTION
**Pull request by kencooke**
### Test Plan

Using Interface from this PR, open the audio selection dialog and verify that each working input device has a VU meter showing realtime audio peak level.

Verify that any glitching/dropouts in the mic audio (when the dialog is open) clear up once the dialog is closed. Test with as many input devices (set to high sample rate eg: 192khz) as possible.

Verify that the audio selection dialog still functions as expected.

### CODE REVIEW

**Comment by kencooke** [comment]
test this please

**Comment by kencooke** [comment]
test this please

**Comment by kencooke** [comment]
test this please

**Comment by kencooke** [comment]
test this please

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1545/HighFidelity-Beta-PR10937-0da19af6f32c274f42a0fc08b18a6bb2cb220280.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1545/HighFidelity-Beta-PR10937-0da19af6f32c274f42a0fc08b18a6bb2cb220280.dmg

**Comment by CainFoool** [comment]
Testing now.

**Comment by OmegaHeron** [comment]
Only offering feedback here - not QA.

It's interesting to note that looking at input devices in desktop mode *without* tablet (i.e.) hud mode shows a device with what looks like much higher input values (clipping into "red" zone) vs tablet that shows same device at about 50% or less of maximal meter space.  Gods of random or is there an inequality in scaling values for hud display meters vs tablet displayed meters?

**Comment by CainFoool** [comment]
# Test Passed
Test | Result
-------- | --------
Using Interface from this PR, open the audio selection dialog and verify that each working input device has a VU meter showing realtime audio peak level. | Pass
Verify that any glitching/dropouts in the mic audio (when the dialog is open) clear up once the dialog is closed. Test with as many input devices (set to high sample rate eg: 192khz) as possible. | Pass
Verify that the audio selection dialog still functions as expected. | Pass

Spent some time testing with multiple audio devices different sample rates and everything was working perfect.

One thing I did notice however and this is currently active in the latest dev build (6853) and Master Build (6850), is that 'Enable Audio Reduction' can't be enabled or disabled - and is off by default. You can hover of it and see it highlight blue, but clicking on it doesn't trigger it.

As this isn't related to a PR, this is a pass however should be fixed in another PR.

Let me know if this needs further testing.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Comment by kencooke** [comment]
@OmegaHeron I think I see the scaling difference you are talking about. But its not the tablet, its the Oculus mic (active when using VR/tablet) only reaching 0.5 at fullscale vs 1.0 for the other mics. This appears to be a bug in the Oculus drivers. You can also see this problem in the Windows Sound Properties dialog which uses the same API we use to obtain peak audio values. 

In the picture below, I am maxing out the capture levels:
image

File: libraries/audio-client/src/AudioPeakValues.cpp [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    return true;
+}
+
+#else
+void AudioClient::checkPeakValues() {
+}
+
+bool AudioClient::peakValuesAvailable() const {
+    return false;
+}
```
Comment (AndrewMeadows) [inline libraries/audio-client/src/AudioPeakValues.cpp]: What? No non-Windows implementation?

Comment (kencooke) [inline libraries/audio-client/src/AudioPeakValues.cpp]: I don't think this feature can be implemented efficiently on the other platforms. 

Using Qt to capture all devices simultaneously (with only 10ms buffering) causes dropouts and performance problems. On Windows we can use a low-level API that doesn't require touching the audio data.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1629/HighFidelity-Beta-PR10937-cf6c1ae4a54391d4d096c5e76805534fbfc60f31.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1629/HighFidelity-Beta-PR10937-cf6c1ae4a54391d4d096c5e76805534fbfc60f31.dmg

**Comment by cozza13** [comment]
I marked this do not merge. Talk to me before it goes in.

**Comment by PhilipRosedale** [comment]
Testing: PASS

This build works super great for me.  I have two inputs and three outputs.  I get correctly displayed levels for both inputs, can switch between all inputs and outputs while on the audio settings tablet page, and can verify loopback without any audio degradation going on and off the audio settings page.  

I would recommend we accept this PR if we can verify good operation on heavily loaded machines/sound cards.  I did find it very helpful to be able to easily see both levels.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2303/HighFidelity-Beta-PR10937-1fc050fdef894751f9888ef97d1c30a4810ab729.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2303/HighFidelity-Beta-PR10937-1fc050fdef894751f9888ef97d1c30a4810ab729.exe

**Pull request merged**